aiida/orm/utils/remote.py

```diff
@@ -69,13 +85,28 @@ def get_calcjob_remote_paths(pks=None, past_days=None, older_than=None, computer
         filters_calc['mtime'] = {'>': timezone.now() - timedelta(days=past_days)}
 
     if older_than is not None:
-        filters_calc['mtime'] = {'<': timezone.now() - timedelta(days=older_than)}
+        older_filter = {'<': timezone.now() - timedelta(days=older_than)}
+        # Check if we need to apply the AND condition
+        if 'mtime' not in filters_calc:
+            filters_calc['mtime'] = older_filter
+        else:
+            past_filter = filters_calc['mtime']
+            filters_calc['mtime'] = {'and': [past_filter, older_filter]}
+
+    if exit_status is not None:
+        filters_calc['attributes.exit_status'] = exit_status
 
     if pks:
         filters_calc['id'] = {'in': pks}
 
+    if only_not_cleaned is True:
+        filters_remote['extras.cleaned'] = {'!==': True}
```

Comment:
Will this match also those that don't even have the extra? As in, should we add an explicit filter `{'!has_key': 'extras.cleaned'}` ? Not sure if this is being tested. Good to check

Reply:
I think is does match to those does not have the extra, but making it more explicity is better.